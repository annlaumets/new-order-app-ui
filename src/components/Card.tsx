import { Box, BoxProps, useStyleConfig } from '@chakra-ui/react'

interface CardHeaderProps extends BoxProps {
  header?: string | React.ReactNode
  children: JSX.Element
  variant?: 'simple'
}

const Card: React.FC<CardHeaderProps> = (props) => {
  const { header, variant, children, ...rest } = props

  const styles = useStyleConfig('Card', { variant })
  return (
    <Box __css={styles} {...rest}>
      {header && (
        <Box borderTopRadius="xl" backgroundColor="gray.85" p="3">
          {header}
        </Box>
      )}
      <Box>{children}</Box>
    </Box>
  )
}

export default Card
