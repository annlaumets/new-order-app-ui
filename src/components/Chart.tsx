import {
  AnimatedGrid,
  Tooltip,
  XYChart,
  AreaSeries,
  buildChartTheme,
  AnimatedAxis,
} from '@visx/xychart'
import { LinearGradient } from '@visx/gradient'
import { Stack, Text, useTheme } from '@chakra-ui/react'
import dayjs from 'dayjs'

import { numberFormatter } from 'lib/utils/format'

import { ChartData, ChartTooltipData } from 'models/chart'

const accessors = {
  xAccessor: (d: ChartData) => {
    // return dayjs(d?.x)?.startOf('day')
    return new Date(`${d?.x}T00:00:00`)
  },
  yAccessor: (d: ChartData) => {
    return d?.y
  },
}

interface LineChartProps {
  data: ChartData[]
}

export const LineChart: React.FC<LineChartProps> = ({ data }) => {
  const theme = useTheme()
  const customTheme = buildChartTheme({
    colors: [theme.colors.brand.pink],
    gridColor: theme.colors.gray[80],
    gridColorDark: theme.colors.gray[80],
    backgroundColor: theme.colors.gray[20],
    tickLength: 1,
  })

  return (
    <XYChart
      key="chart"
      margin={{ left: 40, top: 10, bottom: 20, right: 0 }}
      theme={customTheme}
      height={250}
      xScale={{ type: 'time' }}
      yScale={{ type: 'linear' }}
    >
      <AnimatedGrid
        columns
        numTicks={8}
        lineStyle={{
          stroke: theme.colors.gray[85],
          strokeLinecap: 'round',
          strokeWidth: 1,
        }}
        stroke={theme.colors.gray[85]}
        strokeWidth="1px"
        strokeDasharray="0, 1"
      />
      <LinearGradient
        id="area-gradient"
        from={theme.colors.pink[80]}
        to={theme.colors.pink[90]}
        fromOpacity={0.5}
        toOpacity={0}
      />
      <AreaSeries
        fill="url(#area-gradient)"
        dataKey="supply"
        data={data}
        {...accessors}
      />

      <AnimatedAxis
        orientation="left"
        numTicks={6}
        tickFormat={(value) => {
          return numberFormatter(value)
        }}
        tickLabelProps={() => {
          return { dx: -10 }
        }}
      />

      <AnimatedAxis
        orientation="bottom"
        numTicks={4}
        tickLabelProps={() => {
          return { dx: -10 }
        }}
      />

      <Tooltip<ChartTooltipData>
        key="tooltip"
        showHorizontalCrosshair
        showVerticalCrosshair
        verticalCrosshairStyle={{
          strokeDasharray: '3,5',
          strokeLinecap: 'round',
          stroke: theme.colors.gray[60],
        }}
        horizontalCrosshairStyle={{
          strokeDasharray: '3,5',
          strokeLinecap: 'round',
          stroke: theme.colors.gray[60],
        }}
        renderTooltip={({ tooltipData }) => {
          return (
            <Stack p="1" spacing="2" color="gray.75">
              <Text fontFamily="america" fontWeight="normal" fontSize="0.7rem">
                {dayjs(tooltipData?.nearestDatum?.datum?.x)?.format(
                  'MMM DD, YYYY'
                )}
              </Text>
              <Text fontFamily="america" fontWeight="normal" fontSize="0.7rem">
                {numberFormatter(tooltipData?.nearestDatum?.datum?.y || '')}
              </Text>
            </Stack>
          )
        }}
      />
    </XYChart>
  )
}
