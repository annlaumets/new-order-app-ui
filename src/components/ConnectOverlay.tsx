import { Box } from '@chakra-ui/react'

import WalletWidget from 'components/WalletWidget/WalletWidget'

interface ConnectOverlayProps {
  children: React.ReactNode
  isConnected: boolean
}

const ConnectOverlay: React.FC<ConnectOverlayProps> = ({
  isConnected,
  children,
}) => {
  return (
    <Box pos="relative" borderRadius="md">
      <Box
        pos="absolute"
        top="50%"
        left="50%"
        transform="auto"
        translateX="-50%"
        translateY="-50%"
        zIndex="3"
        display={isConnected ? 'none' : 'block'}
      >
        <WalletWidget />
      </Box>

      <Box
        display={isConnected ? 'none' : 'block'}
        bg="blackAlpha.300"
        backdropFilter="blur(5px)"
        borderRadius="md"
        zIndex="2"
        pos="absolute"
        top="0"
        left="0"
        right="0"
        bottom="0"
      />
      <Box zIndex="1">{children}</Box>
    </Box>
  )
}

export default ConnectOverlay
