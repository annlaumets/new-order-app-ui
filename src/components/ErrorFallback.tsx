import { Box, Button, Container, Flex, Heading } from '@chakra-ui/react'
import Link from 'next/link'
import { FiArrowLeft } from 'react-icons/fi'

interface ErrorFallbackProps {
  error: Error
  resetErrorBoundary: (...args: Array<unknown>) => void
}

export const ErrorFallback: React.FC<ErrorFallbackProps> = () => {
  return (
    <Box as="main" bgColor="black" minH="100vh" overflowX="hidden">
      <Box
        bgImage="url('/images/backgrounds/night-city-error.svg')"
        bgPosition={['left top', 'center top', 'center top']}
        bgRepeat="repeat-x"
        position="relative"
        minH="100vh"
      >
        <Container maxW="container.xl">
          <Flex direction="column" align="center" justify="center" minH="100vh">
            <Heading
              color="green.100"
              fontSize="4xl"
              textTransform="uppercase"
              textAlign="center"
            >
              Something went wrong, please reload or try again later.
            </Heading>
            <Link href="/" passHref>
              <Button
                leftIcon={<FiArrowLeft />}
                variant="outlineGreenBox"
                py="8"
                px="6"
                fontSize="lg"
                mt="8"
              >
                GO BACK TO HOME PAGE
              </Button>
            </Link>
          </Flex>
        </Container>
      </Box>
    </Box>
  )
}
