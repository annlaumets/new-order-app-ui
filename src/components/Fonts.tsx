import { Global } from '@emotion/react'

const Fonts = () => {
  return (
    <Global
      styles={`
      @font-face {
        font-family: 'vectrex';
        font-style: normal;
        font-weight: 700;
        font-display: swap;
        src: url('/fonts/vectrex/bold/Vectrex-Bold.woff') format('woff'), url('/fonts/vectrex/bold/Vectrex-Bold.ttf') format('ttf');
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
      }

      @font-face {
        font-family: 'america';
        font-style: normal;
        font-weight: 400;
        font-display: swap;
        src: url('/fonts/gt-america-mono/regular/GT-America-Mono-Regular.woff') format('woff'), url('/fonts/gt-america-mono/regular/GT-America-Mono-Regular') format('ttf');
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
      }
      `}
    />
  )
}

export default Fonts
