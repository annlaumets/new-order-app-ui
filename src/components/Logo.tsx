import { AspectRatio, AspectRatioProps } from '@chakra-ui/react'
import Image from 'next/image'

const Logo: React.FC<AspectRatioProps> = ({ ...props }) => {
  return (
    <AspectRatio ratio={237 / 56} position="relative" {...props}>
      <Image src="/images/logos/logo.svg" alt="frogs-logo" layout="fill" />
    </AspectRatio>
  )
}

export default Logo
