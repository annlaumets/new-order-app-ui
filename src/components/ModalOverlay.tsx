import { ModalOverlay as ChakraModalOverlay } from '@chakra-ui/react'

const ModalOverlay = () => {
  return <ChakraModalOverlay bg="blackAlpha.300" backdropFilter="blur(5px) " />
}

export default ModalOverlay
