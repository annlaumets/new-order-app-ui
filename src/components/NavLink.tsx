import { Flex, Icon, Link as ChakraLink, Text } from '@chakra-ui/react'
import { useRouter } from 'next/router'
import NextLink from 'next/link'

import { LinkProps } from 'models/sidebarMenu'

import MainMenuMarkerIcon from 'assets/icons/MainMenuMarker.svg'

interface NavLinkProps {
  link: LinkProps
}
const NavLink: React.FC<NavLinkProps> = ({ link, ...rest }) => {
  const router = useRouter()
  const { label, icon, href, isExternal = false } = link
  const currentLocation = router.pathname?.toLowerCase()?.split('/')[1]
  const isActive =
    currentLocation?.length === 0
      ? router.pathname === href
      : href?.includes(currentLocation)

  return (
    <NextLink href={href} passHref>
      <ChakraLink
        isExternal={isExternal}
        href={href}
        _hover={{
          textDecoration: 'none',
        }}
      >
        <Flex
          align="center"
          role="group"
          pl="6"
          pr="4"
          py="1.5"
          backgroundColor={isActive ? 'gray.85' : 'transparent'}
          color={isActive ? 'green.100' : 'white'}
          cursor="pointer"
          _hover={{
            bg: 'gray.60',
            color: 'white',
          }}
          position="relative"
          {...rest}
        >
          {isActive && (
            <Icon
              as={MainMenuMarkerIcon}
              position="absolute"
              top="0"
              left="0"
              h="auto"
            />
          )}
          {icon && (
            <Icon
              mr="4"
              fontSize="16"
              sx={{
                path: {
                  fill: isActive ? 'green.100' : 'gray.40',
                },
              }}
              _groupHover={{
                path: {
                  fill: 'white',
                },
              }}
              as={icon}
            />
          )}
          <Text fontSize="0.95rem">{label}</Text>
        </Flex>
      </ChakraLink>
    </NextLink>
  )
}

export default NavLink
