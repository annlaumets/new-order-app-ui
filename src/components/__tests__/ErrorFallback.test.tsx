import React from 'react'
import { render } from '@testing-library/react'

import { ErrorFallback } from 'components/ErrorFallback'

describe('ErrorFallback.tsx component', () => {
  const { container } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <ErrorFallback error={new Error('test')} resetErrorBoundary={() => {}} />
  )

  test('match to snapshot', () => {
    expect(container).toMatchSnapshot()
  })
})
