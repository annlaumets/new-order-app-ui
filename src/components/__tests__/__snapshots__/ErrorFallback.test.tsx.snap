// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFallback.tsx component match to snapshot 1`] = `
<div>
  <main
    class="css-1sopcer"
  >
    <div
      class="css-1p5gixb"
    >
      <div
        class="chakra-container css-tflc3m"
      >
        <div
          class="css-1jvvll6"
        >
          <h2
            class="chakra-heading css-1cg17vk"
          >
            Something went wrong, please reload or try again later.
          </h2>
          <button
            class="chakra-button css-pqd5kg"
            href="/"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="19"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <polyline
                  points="12 19 5 12 12 5"
                />
              </svg>
            </span>
            GO BACK TO HOME PAGE
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
